crates/blockifier/src/execution/native/syscall_handler.rs

```diff
@@ -0,0 +1,318 @@
+use std::collections::HashSet;
+use std::hash::RandomState;
+
+use cairo_native::starknet::{
+    ExecutionInfo,
+    ExecutionInfoV2,
+    Secp256k1Point,
+    Secp256r1Point,
+    StarknetSyscallHandler,
+    SyscallResult,
+    U256,
+};
+use cairo_vm::vm::runners::cairo_runner::ExecutionResources;
+use starknet_api::core::{ContractAddress, EntryPointSelector};
+use starknet_api::state::StorageKey;
+use starknet_types_core::felt::Felt;
+
+use super::utils::encode_str_as_felts;
+use crate::execution::call_info::{CallInfo, OrderedEvent, OrderedL2ToL1Message};
+use crate::execution::entry_point::{CallEntryPoint, EntryPointExecutionContext};
+use crate::execution::syscalls::hint_processor::OUT_OF_GAS_ERROR;
+use crate::state::state_api::State;
+use crate::transaction::transaction_utils::update_remaining_gas;
+
+pub struct NativeSyscallHandler<'state> {
+    // Input for execution.
+    pub state: &'state mut dyn State,
+    pub execution_resources: &'state mut ExecutionResources,
+    pub execution_context: &'state mut EntryPointExecutionContext,
+
+    // Call information.
+    pub caller_address: ContractAddress,
+    pub contract_address: ContractAddress,
+    pub entry_point_selector: Felt,
+
+    // Execution results.
+    pub events: Vec<OrderedEvent>,
+    pub l2_to_l1_messages: Vec<OrderedL2ToL1Message>,
+    pub inner_calls: Vec<CallInfo>,
+
+    // Additional execution result info.
+    pub storage_read_values: Vec<Felt>,
+    pub accessed_storage_keys: HashSet<StorageKey, RandomState>,
+}
+
+impl<'state> NativeSyscallHandler<'state> {
+    pub fn new(
+        state: &'state mut dyn State,
+        caller_address: ContractAddress,
+        contract_address: ContractAddress,
+        entry_point_selector: EntryPointSelector,
+        execution_resources: &'state mut ExecutionResources,
+        execution_context: &'state mut EntryPointExecutionContext,
+    ) -> NativeSyscallHandler<'state> {
+        NativeSyscallHandler {
+            state,
+            caller_address,
+            contract_address,
+            entry_point_selector: entry_point_selector.0,
+            execution_resources,
+            execution_context,
+            events: Vec::new(),
+            l2_to_l1_messages: Vec::new(),
+            inner_calls: Vec::new(),
+            storage_read_values: Vec::new(),
+            accessed_storage_keys: HashSet::new(),
+        }
+    }
+
+    pub fn execute_inner_call(
```

Comment:
I think since we are not using `execute_inner_call`, `update_remaining_gas` and `substract_syscall_gas_cost` we shouldn't add them just yet